### DISCUSSION TITLE
Multi select selected issue

### DISCUSSION THREAD

**Discussion by somnath-dev-dapl** [area17/twill#2475] [General]
In twill3 I created 2 CRUD, ImportedItem and ItemTypes. I used Multiselect for selecting ItemTypes.
$form->add(
                MultiSelect::make()
                    ->name('item_types_id') 
                    ->label('Item Type')
                    ->inline()
                    ->options(
                        Options::make(
                            ItemType::all()
                                ->map(
                                    fn($sector) => Option::make(
                                        $sector->id,
                                        $sector->title
                                    )
                                )
                                ->toArray()
                        )
                    )
            );
This is the code. I also tried another code also
$form->add(
                MultiSelect::make()
                    ->inline()
                    ->name('item_types_id') 
                    ->label('Item Type') 
                    ->searchable()
                    ->options(
                        app()->make(ItemTypeRepository::class)->listAll()->toArray()
                    )
            );
Here updated on pivot table. But after refresh its not selected.
How do I selected this in multiselect?

![image](https://github.com/area17/twill/assets/115613511/6b670d54-4655-42c4-8a6a-4915d8e599c8)
![image](https://github.com/area17/twill/assets/115613511/87303783-5e9e-4b98-a44d-23f9276370f9)
![image](https://github.com/area17/twill/assets/115613511/7f38215f-6f15-4182-9cc5-a4d8884df18f)

**Comment by ifox** [comment]
Hi @somnath-dev-dapl please share the code you are using in the repository class. It seems like your `afterSave` code is working, but not `getFormFields`, potentially.
